scripts/benchmark_inference.py

```diff
@@ -159,7 +178,21 @@
     torch._C._distributed_c10d._register_process_group("default", dist.group.WORLD)
 
 print("loading model")
-model = models.get_model(args.architecture, args.variant, device_type=args.device_type)
+if args.distributed:
+    distr_param = "tp"
+if args.distributed_strategy:
+    # --distributed_strategy option overrides --distributed option
+    distr_param = args.distributed_strategy
+
+model = models.get_model(
+    args.architecture,
+    args.variant,
+    model_path=args.model_path,
+    device_type=args.device_type,
+    source=args.model_source,
+    distributed_strategy=distr_param,
+    checkpoint_sharding=args.checkpoint_sharding,
+)
```

Comment:
the disadvantage of loading an actual checkpoint into this script is that it can take a while to load. since it doesn't have an effect on the output (we're just measuring time to perform the calculations not verifying quality of results) it might not add value, and might lead users to use the script sub-optimally

Reply:
I see. This benchmark script does not need `source` and `checkpoint_sharding` arguments.